.github/workflows/ci.yml

```diff
@@ -0,0 +1,92 @@
+name: Run checks on pull requests and pushes on main branch
+
+on:
+  pull_request: {}
+  push:
+    branches: [main]
+
+jobs:
+  tests:
+    name: Checks of unit tests and format checks
+    runs-on: "${{ matrix.os }}"
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [3.7, 3.8, 3.9]
```

Comment:
although engineer will only test it locally with a single version (now suggest as 3.7.8), guess still better to test it with any python version > 3.7?

Reply:
I would even recommend adding `3.10` to this list, as it was released on October 4!

Reply:
added